### PR TITLE
Properly fixed pet fake combat and flashing pet attack button

### LINKED ISSUE #5551: [Pet - UI] Fix for "Attack" continues to flash after combat end
Rev: e1b544d129190b921afc5f394379f5df906a7172
OS: XP Sp3
DB: 46 + updates

<b>Description</b>

When you click "attack" on the pet bar, the button continues to flash after combat ends until you log out or dismiss the pet.
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/65193-pet-ui-fix-for-attack-continues-to-flash-after-combat-end?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by ghost**
**Target branch(es):** 3.3.5

- [x] 3.3.5

**Issues addressed:** Closes #5551

**Tests performed:** Compiles, simple ingame tests

Properly fixed pet fake combat, also fixes flashing pet attack button

### CODE REVIEW

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -8547,10 +8551,12 @@ void Unit::SetInCombatState(bool PvP, Unit* enemy)
...
 
-    for (Unit::ControlList::iterator itr = m_Controlled.begin(); itr != m_Controlled.end(); ++itr)
+    for (Unit::ControlList::iterator itr = m_Controlled.begin(); itr != m_Controlled.end();)
     {
-        (*itr)->SetInCombatState(PvP, enemy);
-        (*itr)->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_PET_IN_COMBAT);
+        Unit* controlled = *itr;
+        ++itr;
+
+        controlled->SetInCombatState(PvP, enemy);
```
Comment (Shauren) [inline src/server/game/Entities/Unit/Unit.cpp]: I see you changed iterator incrementing, can SetInCombatState dismiss pet?

Comment (xinef1) [inline src/server/game/Entities/Unit/Unit.cpp]: Don't really remember when i did this, but at some point it time, it certainly could.

**Comment by tkrokli** [comment]
(maybe this is just cosmetic, but I could not resist asking) Wouldn't it be more natural to write the name of the function as `ClearPetInCombat` instead of the current `ClearInPetCombat` ? Or am I missing something?

**Comment by xinef1** [comment]
It may seem strange, but it is more proper if you think about it as type of combat. It is pet combat so we are clearing state in pet combat.

**Pull request merged**